moveit_ros/planning_interface/moveit_cpp/src/moveit_cpp.cpp

```diff
@@ -318,6 +318,7 @@ const std::shared_ptr<tf2_ros::Buffer>& MoveItCpp::getTFBuffer() const
 
 void MoveItCpp::clearContents()
 {
+  tf_listener_.reset();
   tf_buffer_.reset();
   planning_scene_monitor_.reset();
```

Comment:
`planning_scene_monitor_` also maintains a reference(?) to `tf_buffer_` and thus should be deleted afterwards:
https://github.com/ubi-agni/moveit/blob/7005244ca260f6d767086ed29c352c6ea11e02df/moveit_ros/planning_interface/moveit_cpp/src/moveit_cpp.cpp#L127

Reply:
It takes a `shared_ptr` reference unlike the `tf_listener_` which take a reference to the underlying object

Reply:
Nevertheless, it might be cleaner, to reset the PSM before tf_buffer_ as well.